### ISSUE TITLE
[DataGrid] Exclude unused locales from the bundle

### ISSUE THREAD

**Issue by MieleVL** [mui/mui-x#4059]
### Order ID 💳

#33929

### Duplicates

- [X] I have searched the existing issues

### Latest version

- [X] I have tested the latest version

### The problem in depth 🔍

We use i18next to handle internationalization of our applications. mui/x-data-grid-pro locales (localeText) are fetched from the backend, and then passed to `createTheme()`. This means we manage our own translations and thus never import locales from the mui/x-data-grid-pro (or even from the mui/material) javascript library. However, I noticed that our (main) bundle contains all of the locales mui/x-data-grid-pro provides. Is there a way to exclude these locales from our builds (react-scripts v5)?

Below the import statements we use for mui/x-data-grid-pro in different files:

```
import {
    DataGridPro,
    useGridApiRef,
    GridActionsCellItem,
    GRID_CHECKBOX_SELECTION_COL_DEF
} from '@mui/x-data-grid-pro';

import {
    useGridApiContext,
    useGridSelector,
    GridFilterForm,
    gridFilterModelSelector,
    filterableGridColumnsSelector
} from '@mui/x-data-grid-pro';
```

### Your environment 🌎

<details>
  <summary>`npx @mui/envinfo`</summary>

```
  System:
    OS: Windows 10 10.0.22000
  Binaries:
    Node: 14.16.1 - C:\Program Files\nodejs\node.EXE
    Yarn: 1.22.4 - C:\Program Files (x86)\Yarn\bin\yarn.CMD
    npm: 6.14.12 - C:\Program Files\nodejs\npm.CMD
  Browsers:
    Chrome: 98.0.4758.102
    Edge: Spartan (44.22000.120.0), Chromium (98.0.1108.62)
  npmPackages:
    @emotion/react:  11.8.1 
    @emotion/styled:  11.8.1 
    @mui/base:  5.0.0-alpha.69
    @mui/icons-material:  5.4.4
    @mui/lab:  5.0.0-alpha.70
    @mui/material:  5.4.4
    @mui/private-theming:  5.4.4
    @mui/styled-engine:  5.4.4
    @mui/system:  5.4.4
    @mui/types:  7.1.2
    @mui/utils:  5.4.4
    @mui/x-data-grid-pro:  5.5.1
    @mui/x-license-pro:  5.5.1
    @types/react:  17.0.39
    react:  17.0.2
    react-dom:  17.0.2
```
</details>

**Comment by flaviendelangle** [comment]
#3965 should enable automatic tree shaking in webpack and fix your problem.
You will only bundle the default locale (english).

**Comment by MieleVL** [comment]
I can confirm that after using the package versions mentioned above, the locales are no longer included!
Thanks! Appreciate the hard work!

**Comment by flaviendelangle** [comment]
Note that it is a temporary version linked to a specific commit
The real release will be `5.6.0` at the end of this week.

**Comment by flaviendelangle** [comment]
Fixed by #4059

**Comment by MieleVL** [comment]
Yes, thanks for the heads up.
